### ISSUE TITLE
Different AUC using sklearn and catboost

### ISSUE THREAD

**Issue by Paul0M** [catboost/catboost#1108]
Problem: Different AUC using sklearn and catboost. Maybe there is something wrong in the inner implementation of catboost , for cbu.get_roc_curve? See 3 comparison bellow. @annaveronika 
catboost version: 0.20
Operating System: ubuntu 1604
CPU:
[https://github.com/catboost/tutorials/blob/master/classification/classification_with_parameter_tuning_tutorial.ipynb](https://github.com/catboost/tutorials/blob/master/classification/classification_with_parameter_tuning_tutorial.ipynb)
1. using below lines, auc will be 0.92+, original in the tutorial
```python
fpr, tpr, _ = cbu.get_roc_curve(model, dataset, plot=True)
auc = sklearn.metrics.auc(fpr, tpr)
print(auc)
# 0.9236684738800891
```
2. But when using these lines, auc is just 0.79+
```python
y_pred = model.predict(X_test)
from sklearn.metrics import roc_auc_score
roc_auc_score(y_test, y_pred)
# 0.7927506653945289
```
3. when using predict_prob, and using column 1, the results matches
```python
yy = model.predict_proba(X_test)
print(roc_auc_score(y_test, yy[:,0]),roc_auc_score(y_test, yy[:,1]))
# 0.0763315261199109, 0.9236684738800891
```

![auc](https://user-images.githubusercontent.com/13173000/70585978-d8236f00-1c00-11ea-9cc3-bdaa9d077f16.png)

![auc2](https://user-images.githubusercontent.com/13173000/70589213-1f622d80-1c0a-11ea-8033-ef066e952f1c.png)

**Comment by LyzhinIvan** [comment]
Hi, @Paul0M ! Why do you think that something is wrong? ROC-AUC is always calculated with probabilities, not binarized labels. So get_roc_curve calls predict_proba inside.

**Comment by Paul0M** [comment]
@LyzhinIvan Sorry, my mistake. Thanks for the info.
